### PR TITLE
Events

### PR DESCRIPTION
**Pull request by glipR**
Addresses https://github.com/MelbourneHighSchoolRobotics/ev3sim/issues/84.

### CODE REVIEW

File: ev3sim/attach.py [hunk 1]
```diff
@@ -442,7 +453,7 @@ def close(self):
                 @mock.patch("ev3sim.code_helpers.is_sim", True)
                 @mock.patch("ev3sim.code_helpers.CommServer", MockedCommServer)
                 @mock.patch("ev3sim.code_helpers.CommClient", MockedCommClient)
-                @mock.patch("ev3sim.code_helpers.wait_for_tick", wait_for_tick)
```
Comment (peter-drew) [inline ev3sim/attach.py:445]: This doesn't look right

File: ev3sim/code_helpers.py [hunk 2]
```diff
@@ -57,6 +57,22 @@ def close(self):
         self.socket.close()
 
 
-def wait_for_tick():
-    """If in simulation, waits for the next simulation tick."""
-    pass
```
Comment (peter-drew) [inline ev3sim/code_helpers.py:62]: Seems my quick merge after black ran did not quite work out

File: ev3sim/presets/soccer.py [hunk 3]
```diff
@@ -127,9 +128,10 @@ def reset(self):
         self.resetPositions()
         self.time_tick = 0
         self.update_time_text = True
+        for robotID in ScriptLoader.instance.robots.keys():
+            ScriptLoader.instance.sendEvent(robotID, GAME_RESET, {})
 
     def resetPositions(self):
-        # It is assumed that 2 robots to each team, with indexes increasing as we go across teams.
```
Comment (peter-drew) [inline ev3sim/presets/soccer.py:132]: Here as well

File: ev3sim/robots/demo.py [hunk 4]
```diff
@@ -10,7 +10,15 @@
 from ev3dev2.motor import LargeMotor
 from ev3dev2.sensor.lego import ColorSensor, UltrasonicSensor
 from ev3dev2.sensor import Sensor
-from ev3sim.code_helpers import is_sim, wait_for_tick
```
Comment (peter-drew) [inline ev3sim/robots/demo.py:13]: and here

File: ev3sim/single_run.py [hunk 5]
```diff
@@ -16,13 +16,14 @@ def single_run(preset_filename, robots, bind_addr):
     config["robots"] = config.get("robots", []) + robots
 
     shared_data = {
```
Comment (peter-drew) [inline ev3sim/single_run.py:18]: not sure about this either

**Review by peter-drew — CHANGES_REQUESTED** [review]

File: ev3sim/presets/soccer.py (hunk 3 above)
Comment (glipR) [reply ev3sim/presets/soccer.py:132]: Removed this as it isn't really true anymore.

File: ev3sim/single_run.py (hunk 5 above)
Comment (glipR) [reply ev3sim/single_run.py:19]: Also just happened to do this while adding the "events" key, figured we should have full stops on all of these comments.

**Review by peter-drew — APPROVED** [review]

**Pull request merged**